lib/service/storage/elasticsearch.js

```diff
@@ -822,6 +822,71 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+   * Delete fields of a document and replace it
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Array}  fields - Document fields to be removed
+   * @param {Object} options - refresh (undefined), userId (null)
+   *
+   * @returns {Promise.<{ _id, _version, _source }>}
+   */
+  async deleteFields (
+    index,
+    collection,
+    id,
+    fields,
+    { refresh, userId=null } = {})
+  {
+    const esIndex = this._getESIndex(index, collection);
+    const esRequest = {
+      id,
+      index: esIndex,
+    };
+
+    // metadata
+    const _kuzzle_info = {
+      author: getUserId(userId),
```

Comment:
The `author` property should not be updated since the `deleteFields` action is considered as an update.